lib/discover_reddit_bot/text_formatter.ex

```diff
@@ -41,24 +44,58 @@ defmodule DiscoverRedditBot.TextFormatter do
       id: "all",
       title: "Share all subreddits found",
       input_message_content: %InputTextMessageContent{
-        message_text: format_subreddits(subreddits, urls),
+        message_text: format_subreddits(url_subreddits),
         parse_mode: "Markdown"
       }
     }
 
     [all_article] ++ articles
   end
 
+  @spec no_subreddits_text :: String.t()
+  def no_subreddits_text() do
+    "No subreddits detected"
+  end
+
+  @spec get_no_subreddits_inline :: [InlineQueryResultArticle.t()]
   def get_no_subreddits_inline() do
+    text = no_subreddits_text()
+
     [
       %InlineQueryResultArticle{
         type: "article",
         id: "nosubs",
-        title: "No subreddits detected",
+        title: text,
         input_message_content: %InputTextMessageContent{
-          message_text: "No subreddits detected"
+          message_text: text
         }
       }
     ]
   end
+
+  defp merge_subreddits(all) do
+    all
+    |> Enum.reduce(%{}, fn {url, subreddits}, merged ->
+      Enum.reduce(subreddits, merged, fn {sub, n}, merged ->
+        {current, urls} = Map.get(merged, sub, {0, []})
+
+        data = {current + n, urls ++ [url]}
+        Map.put(merged, sub, data)
+      end)
+    end)
+    |> Enum.map(fn {sub, {n, urls}} -> {sub, n, urls} end)
+  end
+
+  defp format_subreddit({sub, n}), do: "  - [#{sub}](#{reddit_base_url()}#{sub}) (#{n})"
+
+  defp format_subreddit({sub, n, urls}, :for_inline) do
```

Comment:
We probably don't need the `:for_inline` anymore don't we?

Reply:
Nope, I'll remove it 👌